core/src/com/unciv/logic/automation/unit/CityLocationTileRanker.kt

```diff
@@ -1,143 +1,141 @@
-                        .filter { canUseTileForRanking(it, unit.civ) }
-                        .filter { it.getContinent() == center.getContinent() }
-                )
-                yieldAll(
-                    center.getTilesInDistance(modConstants.minimalCityDistanceOnDifferentContinents)
-                        .filter { canUseTileForRanking(it, unit.civ) }
-                        .filter { it.getContinent() != center.getContinent() }
-                )
-            }
-        }.toSet()
-
-        // This is to improve performance - instead of ranking each tile in the area up to 19 times, do it once.
-        val nearbyTileRankings = getNearbyTileRankings(unit.getTile(), unit.civ)
-
-        val distanceFromHome = if (unit.civ.cities.isEmpty()) 0
-        else unit.civ.cities.minOf { it.getCenterTile().aerialDistanceTo(unit.getTile()) }
-        val range = (8 - distanceFromHome).coerceIn(1, 5) // Restrict vision when far from home to avoid death marches
+    /**
+     * Returns a hashmap of tiles to their ranking plus the a the highest value tile and its value
+     */
+    fun getBestTilesToFoundCity(unit: MapUnit, distanceToSearch: Int? = null): Pair<HashMap<Tile, Float>, Pair<Tile?, Float>> {
+        val range =  if (distanceToSearch != null) distanceToSearch else {
+            val distanceFromHome = if (unit.civ.cities.isEmpty()) 0
+            else unit.civ.cities.minOf { it.getCenterTile().aerialDistanceTo(unit.getTile()) }
+            (8 - distanceFromHome).coerceIn(1, 5) // Restrict vision when far from home to avoid death marches
+        }
+        val nearbyCities = unit.civ.gameInfo.getCities()
+            .filter { it.getCenterTile().aerialDistanceTo(unit.getTile()) <= 7 + range }
 
         val possibleCityLocations = unit.getTile().getTilesInDistance(range)
-            .filter { canUseTileForRanking(it, unit.civ) }
-            .filter {
-                val tileOwner = it.getOwner()
-                it.isLand && !it.isImpassible() && (tileOwner == null || tileOwner == unit.civ) // don't allow settler to settle inside other civ's territory
-                    && (unit.currentTile == it || unit.movement.canMoveTo(it))
-                    && it !in tilesNearCities
+            .filter { canSettleTile(it, unit.civ, nearbyCities) && (unit.getTile() == it || unit.movement.canMoveTo(it)) }
+        val uniqueCache = LocalUniqueCache()
+        val baseTileMap = HashMap<Tile, Float>()
+        val tileRankMap = HashMap<Tile, Float>()
+        var maxValueTile: Tile? = null
+        var maxValueRank: Float = 0f
+        for (tile in possibleCityLocations) {
+            val tileValue = rankTileToSettle(tile, unit.civ, nearbyCities, baseTileMap, uniqueCache)
+            if (tileValue > maxValueRank) {
+                maxValueTile = tile
+                maxValueRank = tileValue
             }
+            tileRankMap[tile] = tileValue
+        }
+        return Pair(tileRankMap, Pair(maxValueTile, maxValueRank))
```

Comment:
Should be a data class, instead of a triplet - Would make things clearer on both ends for the next person reading this